spec/ruby/core/queue/timeout_conversion_spec.rb

```diff
@@ -0,0 +1,6 @@
+require_relative '../../spec_helper'
+require_relative '../../shared/types/rb_num2dbl_fails'
+
+describe "Queue operations with timeout" do
+  it_behaves_like :rb_num2dbl_fails, nil, -> timeout { q = Queue.new; q.push(1); -> { q.pop(timeout: timeout) } }
+end
```

Comment:
Is `pop` intentionally hardcoded. It will be nice to have it called for each `pop` alias method.

Reply:
Actually as far as it becomes complicated - I would propose to extract these tests for the `timeout` parameter coercion into a separate branch/PR to not delay merging this PR.

Reply:
> Is pop intentionally hardcoded? It will be nice to have it called for each pop alias method.

No, it was missing from the main test so I left it limited - though it's a simple fix, so I added it.

> suggestion: It's should be located in a method spec file or shared example, not in a separate file.

Moved it into `pop_spec.rb`.

> to use it_should_behave_like method instead of it_behaves_like and

I looked it from a bunch of angles and couldn't figure out how it would work. If you strongly believe in using it, do you have an example implementation for this?

Reply:
> Actually as far as it becomes complicated - I would propose to extract these tests for the timeout parameter coercion into a separate branch/PR to not delay merging this PR.

Please look at the latest version. I consider it solid and flexible enough. If there is an issue, please let me know and I can still separate it.

Reply:
I like the new shared example!

Reply:
Just as a side note.

>>to use it_should_behave_like method instead of it_behaves_like and

> I looked it from a bunch of angles and couldn't figure out how it would work. If you strongly believe in using it, do you have an example implementation for this?

`it_should_behave_like` is more low level method (it is used by `it_behaves_like` under the hood) and doesn't have `method`/`object` arguments.

When it's used inside a shared example it _sees_ instance variables specified by the outer shared example (that is usually is called with `it_behaves_like`, e.g. `@method` and `@object`).

Let's look at the shared example `array_select` (`spec/ruby/core/array/shared/select.rb`).

```ruby
describe :array_select, shared: true do
  it_should_behave_like :enumeratorize
```

```ruby
describe :enumeratorize, shared: true do
  it "returns an Enumerator if no block given" do
    [1,2].send(@method).should be_an_instance_of(Enumerator)
  end
end
```

So `enumeratorize` has access to the `@method` used in the outer example `array_select`.

In case we need some additional information - a new instance variable could assigned and become visible inside a nested shared example:

```ruby
describe "Array#all?" do
  @value_to_return = -> _ { true }
  it_behaves_like :array_iterable_and_tolerating_size_increasing, :all?
```

Reply:
This makes sense. I missed a few things including the fact that it can be called within and outside of a `it_behaves_like` context. Thanks for the details! I think because of the pattern we need here (transferring a 1-param lambda to catch an exception would be a stretch for this).